# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../xfce4-settings/fix_xi.patch
# Copyright (C) 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -upr xfce4-settings-4.6.1/dialogs/mouse-settings/main.c xfce4-settings-4.6.1_new/dialogs/mouse-settings/main.c
--- xfce4-settings-4.6.1/dialogs/mouse-settings/main.c	2009-03-28 10:57:59.000000000 +0100
+++ xfce4-settings-4.6.1_new/dialogs/mouse-settings/main.c	2009-12-15 08:55:58.292245950 +0100
@@ -60,6 +60,11 @@
 #define IsXExtensionPointer 4
 #endif
 
+/* Xi 1.4 is required */
+#define MIN_XI_VERS_MAJOR 1
+#define MIN_XI_VERS_MINOR 4
+
+
 /* settings */
 #ifdef HAVE_XCURSOR
 #define PREVIEW_ROWS    (3)
@@ -1221,15 +1226,24 @@ main (gint argc, gchar **argv)
 
         return EXIT_FAILURE;
     }
-    
-    /* check for Xi 1.4 */
+   
+    /* check for Xi */
     version = XGetExtensionVersion (GDK_DISPLAY (), INAME);
-    if (!version || !version->present || version->major_version < 1 || version->minor_version < 4)
+    if (version == NULL || !version->present)
+    {
+        g_critical ("XI is not present.");
+        return EXIT_FAILURE;
+    }
+    else if (version->major_version < MIN_XI_VERS_MAJOR
+             || (version->major_version == MIN_XI_VERS_MAJOR
+                 && version->minor_version < MIN_XI_VERS_MINOR))
     {
-        g_critical ("XI is not present or too old.");
-        
+        g_critical ("Your XI is too old (%d.%d) version %d.%d is required.",
+                    version->major_version, version->minor_version,
+                    MIN_XI_VERS_MAJOR, MIN_XI_VERS_MINOR);
         return EXIT_FAILURE;
     }
+ 
 
     /* open the xsettings and pointers channel */
     xsettings_channel = xfconf_channel_new ("xsettings");
diff -upr xfce4-settings-4.6.1/xfce4-settings-helper/pointers.c xfce4-settings-4.6.1_new/xfce4-settings-helper/pointers.c
--- xfce4-settings-4.6.1/xfce4-settings-helper/pointers.c	2009-03-28 10:57:59.000000000 +0100
+++ xfce4-settings-4.6.1_new/xfce4-settings-helper/pointers.c	2009-12-15 09:16:23.422607095 +0100
@@ -42,6 +42,10 @@
 
 #define MAX_DENOMINATOR (100.00)
 
+/* Xi 1.4 is required */
+#define MIN_XI_VERS_MAJOR 1
+#define MIN_XI_VERS_MINOR 4
+
 /* test if the required version of inputproto (1.4.2) is available */
 #if XI_Add_DevicePresenceNotify_Major >= 1 && defined (DeviceRemoved)
 #define HAS_DEVICE_HOTPLUGGING
@@ -137,14 +141,18 @@ xfce_pointers_helper_init (XfcePointersH
     /* query the extension version */
     version = XGetExtensionVersion (xdisplay, INAME);
     
-    /* check for Xi 1.4 */
-    if (!version || !version->present || version->major_version < 1 || version->minor_version < 4)
+    /* check for Xi */
+    if (version == NULL || !version->present)
     {
-        /* print error */
-        g_critical ("XI is not present or too old.");
-
-        /* no channel */
-        helper->channel = NULL;
+        g_critical ("XI is not present.");
+    }
+    else if (version->major_version < MIN_XI_VERS_MAJOR
+             || (version->major_version == MIN_XI_VERS_MAJOR
+                 && version->minor_version < MIN_XI_VERS_MINOR))
+    {
+        g_critical ("Your XI is too old (%d.%d) version %d.%d is required.",
+                    version->major_version, version->minor_version,
+                    MIN_XI_VERS_MAJOR, MIN_XI_VERS_MINOR);
     }
     else
     {
